reactor-core/src/test/java/reactor/core/publisher/FluxRetryWhenTest.java

```diff
@@ -58,6 +58,33 @@ public class FluxRetryWhenTest {
 	Flux<Integer> rangeError = Flux.concat(Flux.range(1, 2),
 			Flux.error(new RuntimeException("forced failure 0")));
 
+	@Test
+	//https://github.com/reactor/reactor-core/issues/3253
+	public void shouldFailWhenOnErrorContinueEnabled() {
+		Mono.create(sink -> {
+					throw new RuntimeException("blah");
+				})
+				.retryWhen(Retry.indefinitely().filter(t -> false))
+				.onErrorContinue((e, o) -> {})
+				.as(StepVerifier::create)
+				.expectError()
+				.verify(Duration.ofSeconds(10));
+	}
+
+	@Test
+	//https://github.com/reactor/reactor-core/issues/3253
+	public void shouldWorkAsExpected() {
+		Mono.just(1)
+			.map(v -> { // ensure original context is propagated
```

Comment:
What do you mean by the comment? Is there any implicit validation of context preservation happening?